geoportailv3/mymaps.py

```diff
@@ -0,0 +1,321 @@
+# -*- coding: utf-8 -*-
+import uuid
+import datetime
+import sqlahelper
+
+from sqlalchemy import Column, ForeignKey, Table
+from sqlalchemy.types import Unicode, Boolean, DateTime, Integer, Float, Binary
+from sqlalchemy.orm import relationship, sessionmaker
+from sqlalchemy.ext.declarative import declarative_base
+
+
+from shapely import wkb
+from shapely.geometry import asShape
+
+from shapely.geometry.multipoint import asMultiPoint
+from shapely.geometry.multilinestring import asMultiLineString
+from shapely.geometry.multipolygon import MultiPolygonAdapter
+from geoalchemy2 import Geometry
+from geoalchemy2.shape import from_shape
+
+import geojson
+
+engine = sqlahelper.get_engine('mymaps')
+Base = declarative_base(bind=engine)
+Session = sessionmaker(bind=engine)
+DBSession = Session()
+
+
+class Map(Base):
+    __tablename__ = 'map'
+    uuid = Column(Unicode, primary_key=True,
+                  default=lambda: unicode(uuid.uuid4().hex))
+    user_login = Column(Unicode(50))
+    title = Column(Unicode(50))
+    description = Column(Unicode)
+    public = Column(Boolean)
+    create_date = Column(DateTime, default=datetime.datetime.now)
+    update_date = Column(DateTime, onupdate=datetime.datetime.now)
+    zoom = Column(Integer)
+    x = Column(Float)
+    y = Column(Float)
+    theme = Column(Unicode)
+    bg_layer = Column(Unicode)
+    bg_opacity = Column(Float, default=100)
+    layers = Column(Unicode)
+    layers_indices = Column(Unicode)
+    layers_opacity = Column(Unicode)
+    layers_visibility = Column(Unicode)
+    selected_node = Column(Unicode)
+    rating = Column(Float, default=0)
+    rating_count = Column(Integer, default=0)
+    category_id = Column(Integer, ForeignKey('category.id'))
+    label = Column(Unicode)
+    category = relationship('Category', backref='maps')
+    features = relationship('Feature', backref='map')
+
+    def get_title(self):
+        if self.title is not None:
+            return self.title.replace("'", "_")
+        return None
+
+    def todict(self):
+        def convert_datetime(value):
+            if value is not None:
+                return value.strftime("%Y-%m-%d %H:%M:%S")
+            else:
+                return None
+
+        for c in self.__table__.columns:
+            if isinstance(c.type, DateTime):
+                value = convert_datetime(getattr(self, c.name))
+            else:
+                value = getattr(self, c.name)
+            yield(c.name, value)
+
+    def __iter__(self):
+        """Returns an iterable that supports .next()
+            so we can do dict(sa_instance)
+        """
+        return self.todict()
+
+    @staticmethod
+    def get(id):
+        """ Get map by its id. """
+        return DBSession.query(Map).get(id)
```

Comment:
I see that you do the same in `belonging_to`. So let's forget about this. This is minor anyway.